### DISCUSSION TITLE
What's the difference between useMutation's side effects and mutate's side effects

### DISCUSSION THREAD

**Discussion by json9512** [TanStack/query#3231] [Q&A] [answered]
Hi,
I've read these posts ([post1](https://tkdodo.eu/blog/mastering-mutations-in-react-query), [post2](https://react-query.tanstack.com/guides/mutations#mutation-side-effects)) and got curious about the difference between the two.

From what I understood, 
- `useMutation`'s side effects (eg. `onSuccess`) run before `mutate`'s side effects
- `mutate`'s side effects might not run if the component unmounts before it is called

Could you please provide some examples where I would need to use `mutate`'s side effects over `useMutation`'s side effects?
(Some examples of `trigger component-specific side effects` cases will be appreciated) 🙏  🙏

**Comment by TkDodo** [comment, marked as answer]
route changes or showing toast notifications for example. everything that you'd like to do that is irrelevant if the user has already moved to a different screen before the mutation is finished.
